extensions/sql-migration/src/scripts/SHIR-auto-configuration.ps1

```diff
@@ -0,0 +1,609 @@
+#>
+
+param (
+	[switch] $AdminPriv = $false,
+	[string] $AuthKey1 = $null,
+	[string] $AuthKey2 = $null
+)
+
+$timeStamp = [System.DateTime]::Now.ToString("yyyyMMddHHmmss")
+# Unique script id for every run - Telemetry purpose
+$Global:ScriptId = "Script-$timestamp"
+
+# TODO: Auto populate
+$Global:LatestIRVersion = [Version]"5.35.8686.1"
+
+# Minimum NuGet version required to install Az.DataMigration module
+$Global:MinRequiredNuGetVersion = [Version]"2.8.5.201"
+
+# Log file path
+$Global:Logfile = [System.IO.Path]::Combine($env:USERPROFILE, "shir", "shir-$timeStamp.log")
+
+# Guid used to create a script lock (to disable concurrent script instances)
+$Global:InstallationLockGUID = $MyInvocation.MyCommand.Name
+
+################################### Main Script Block ##################################
+$main = {
+	$isFirstInstance = $false
+	# Create a new Stopwatch object
+	$stopwatch = New-Object System.Diagnostics.Stopwatch
+	# Start the stopwatch
+	$stopwatch.Start()
+
+	try {
+		New-Item -ItemType File -Path $Global:Logfile -Force | Out-Null
+		# Create a lock file so that only one instance of script can run at a time
+		$isFirstInstance = Lock-ScriptInstance
+		if ($isFirstInstance) {
+			Install-IR
+		}
+	}
+	catch {
+		Write-ErrorAndLog -exception $_.Exception
+		Write-ErrorAndLog "The script could not configure SHIR due to an internal exception. Please try again."
+	}
+	finally {
+		# Only allow the first running instance to unlock the script
+		if ($isFirstInstance) {
+			Unlock-ScriptInstance
+		}
+		# Stop the stopwatch
+		$stopwatch.Stop()
+
+		# Write the total time taken
+		Write-OutputAndLog ("Total time taken: {0}" -f $stopwatch.Elapsed)
+		Write-OutputAndLog ("Log file generated: " + $Global:Logfile)
+	}
+}
+
+################################### Download, install and configure IR ##################################
+Function Install-IR {
+	Write-OutputAndLog "Checking if Integration Runtime is already installed on this machine..."
+	$installedIRVersion = Get-InstalledShirVersion
+
+	if ($null -ne $installedIRVersion) {
+		Write-OutputAndLog "Found Integration Runtime installed on the machine."
+
+		# Perform validation checks for installed IR
+		if ($installedIRVersion -ne $Global:LatestIRVersion) {
+			Write-ErrorAndLog "Installed Integration Runtime does not meet the checks for configuring. Please read the documentation for more information."
+			return
+		}
+		Write-OutputAndLog "Installed Integration Runtime meets the requirements for configuring."
+
+		if (-not (Test-InternetConnectivity)) {
+			return
+		}
+
+		# Configure SHIR - Register IR with the auth key
+		Register-IntegrationRuntime -AuthKey1 $AuthKey1 -AuthKey2 $AuthKey2
+	}
+	elseif (Test-CanInstallIR) {
+		$downloadFolder = Join-Path (New-Object -ComObject Shell.Application).NameSpace('shell:Downloads').Self.Path "SHIR-Installer"
+		$downloadUrl = "aka.ms/downloadLatestIR"
+
+		# Download the installer
```

Comment:
Consider individual times for each operation - e.g. how long it took for download, vs install vs configure

Reply:
Added logs to time download of msi, installation of msi and installation of required modules